android/app/src/main/java/com/chorestory/app/ChildQuestDetailsActivity.java

```diff
@@ -48,60 +65,94 @@ protected void onCreate(Bundle savedInstanceState) {
             statusImage = R.drawable.yellow_check_mark;
             addProofButton.setVisibility(addProofButton.GONE);
-        } else if (questStatus == QuestCompletion.COMPLETED) {
+        } else if (status == QuestCompletion.COMPLETED) {
             statusImage = R.drawable.green_check_mark;
             addProofButton.setVisibility(addProofButton.GONE);
         } else {
             statusImage = R.drawable.red_check_mark;
         }
         statusImageView.setImageResource(statusImage);
 
-        // Date
-        dateTextView.setText(getDateText());
+        // Date and Time
+        Calendar cal = Calendar.getInstance();
+        cal.setTimeInMillis((long) quest.getDueDate() * 1000);
 
-        // Time
+        mYear = cal.get(Calendar.YEAR);
+        mMonth = cal.get(Calendar.MONTH);
+        mDay = cal.get(Calendar.DAY_OF_MONTH);
+        mHour = cal.get(Calendar.HOUR_OF_DAY);
+        mMinute = cal.get(Calendar.MINUTE);
+
+        dateTextView.setText(getDateText());
         timeTextView.setText(getTimeText());
 
         // Recurrence
-        recurrenceTextView.setText("Repeat once a week");
+        recurrenceTextView.setText(quest.getRecurrenceText());
 
-        // TODO: fetch actual value
-        questDescriptionTextView.setText("This is the description of the quest");
+        // Description
+        questDescriptionTextView.setText(quest.getDescription());
+
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        String token = tokenHandler.getToken(this);
+        if (token != null) {
+            if (tokenHandler.isChildToken(token)) {
+                int qid = getIntent().getIntExtra(getResources().getString(R.string.qid), 1);
+                int ts = getIntent().getIntExtra(getResources().getString(R.string.ts), -1);
+                Call<GetQuestResponse> getQuestQuery;
+                if (ts == -1) {
+                    getQuestQuery = retrofitInference.get_quest(token, qid);
+                } else {
+                    ts -= 60 * 5; // the response uses math.round so subtract 5 minutes to be safe
+                    getQuestQuery = retrofitInference.get_quest_ts(token, qid, ts + ".0");
+                }
+                getQuestQuery.enqueue(new Callback<GetQuestResponse>() {
+                    @Override
+                    public void onResponse(Call<GetQuestResponse> call, Response<GetQuestResponse> response) {
+                        if (response.isSuccessful() && response.body() != null && response.body().hasResponse()) {
+                            quest = new QuestRecyclerViewItem(response.body().getQuest());
+                            populateFields();
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<GetQuestResponse> call, Throwable t) {
+                        // TODO: delete the token we have stored and redirect the user to the login page?
```

Comment:
call `deleteTokenNavigateMain` once Cristian's PR is merged

Reply:
this method is only for fragments?